currency.py

```diff
@@ -8,71 +8,72 @@ def EXIT(message, exit_code):
     sys.exit(exit_code)
 
 
-def date_check(date):    #we just need to check if date in range [1992-07-01;<today>] dmy
+def date_check(dateinp):
+    min_date = datetime(1992, 7, 1)
+    max_date = datetime.today()
 
-    lst[0] = int(lst[0])
-    lst[1] = int(lst[1])
-    lst[2] = int(lst[2])
-    today = datetime.today()
-
-    if date[0] > today.day and date[2] == today.year and date[1] == today.month:
-        return False
-
-    if (date[1] > today.month and date[2] == today.year) or (date[2] == 1992 and date[1] < 7):
-        return False
-
-    if date[2] < 1992 or date[2] > today.year:
-        return False
-
-    return True
+    if min_date <= dateinp <= max_date:
+        return True
+    return False
 
 
 EXIT_SUCCESS = 0
 CURRENCY_ERROR = 1
 DATE_ERROR = 2
 EXIT_REFERENCE = 3
 INPUT_ERROR = 4
+rFlag = False
 
 if len(sys.argv) == 1 or sys.argv[1] == "-h" or sys.argv[1] == "-a":
     print(
         "usage: currency.py [valute code] [date: <format{yyyy-mm-dd}>]", "\n",
         "usage example: eur 2020-01-01", "\n",
         "this script has data from 1992-07-01", "\n",
-        "other arguments: [-h] [-a] [-ec] [-exit_codes]", sep=""
+        "in default program show direct quotations", "-n"
+        "other arguments: [-h] [-a] [-e] [--exit-codes] [-r] [--reverse-quotation] [man]", sep=""
           )
-    EXIT("", EXIT_REFERENCE)
+    EXIT("help shown successful", EXIT_REFERENCE)
 elif sys.argv[1] == "--exit-codes" or sys.argv[1] == "-e":
     print(
         "exit code 0: success", "\n",
         "exit code 1: currency-code format error", "\n",
         "exit code 2: date error(not in data base or date in future)", "\n",
         "exit code 3: references", sep=""
     )
-    EXIT("", EXIT_REFERENCE)
-
-if len(sys.argv) != 3:
-    EXIT("Input should have format: <currency code> <date>.", INPUT_ERROR)
+    EXIT("help shown successful", EXIT_REFERENCE)
+elif sys.argv[1] == "man":
```

Comment:
While it's good that you've thought about `man`, but `man` is a different beast. Being a system-wide command, `man` requires separate files to be created in order to represent man pages (more info [here](https://www.cyberciti.biz/faq/linux-unix-creating-a-manpage/)). So I recommend skipping this `man` argument and move all the text under the "usage" section